### DISCUSSION TITLE
Make it easier to uncheck/deselect a particular cache in the Clear Caches tool

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9447] [Ideas] [answered]
> Created by: Mats Mikkel Rummelhoff (mail@mmikkel.no) on 2015/03/09 00:12:31 +0000
> Votes at time of UserVoice import: 2

-----------------------------------

Having to clear all caches except one or two with the Clear Caches tool is a chore.

Initially, all the caches are checked, but there's no way to easily uncheck or deselect a particular cache – whenever you need to clear all caches except one or two, you'll have to deselect ALL the caches, then select all the ones you want to clear.

It'd be great if the Clear Caches tool worked a little bit more like the user permissions' check boxes – keep the "Deselect/Select all" option, but make it easy to uncheck a particular cache without having to click 10 times :)

**Comment by jrrdnx** [comment]
Came here to start this exact discussion. Would love to be able to check ALL boxes and then selectively uncheck the 2-3 I don't want to clear, rather than have to check them all again one by one. But currently if the ALL box is checked you're blocked from unchecking any others.

Not sure if there's any accessibility implications with the current functionality.

<img width="360" height="563" alt="Screenshot 2025-12-11 at 10 12 02" src="https://github.com/user-attachments/assets/5142a7e1-1669-4883-955d-43411e0e1d34" />

**Reply by mmikkel** [reply]
I was going to suggest the [CP Clear Cache](https://plugins.craftcms.com/cp-clearcache?craft5) plugin, which I originally wrote after making this issue... ten years ago 🫠

CP Clear Cache does have the same behaviour where deselecting _All_ does uncheck all the checkboxes, but my approach to the problem was to have it remember which boxes you've checked (via `localStorage`) – as in my case I tend to clear the same things most of the time. Which, at least in my case, mostly negates the issue 😊

**Comment by brandonkelly** [comment, marked as answer]
Just updated the Caches utility for Craft 5.9, so that it remembers which cache options were previously selected. (8d7b5d767c95a3553185ad776a3318476cb151c2)

**Reply by brandonkelly** [reply]
Craft 5.9.0 is out with that change 🎉
